subprojects/signing/src/main/java/org/gradle/plugins/signing/Signature.java

```diff
@@ -309,8 +302,15 @@ private Date defaultDate() {
         return new Date(modified);
     }
 
-    public void setFile(File file) {
-        this.file = file;
+    /**
+     * Set the file for the generated signature.
+     *
+     * @param file ignored
+     * @deprecated changing the output file is not supported.
+     */
+    @Deprecated
+    public void setFile(@SuppressWarnings("unused") File file) {
+        // do nothing
```

Comment:
❌ We should print a deprecation warning when this method is used. Did you check how the `file` property shows up in the DSL docs? It should show up as not deprecated.

Reply:
❌ We also need to add a section to the release notes about possibly breaking changes.

Reply:
> We should print a deprecation warning when this method is used.

Done in 62d13aecca547272b44f5797e6b5df49c53231b2.

> Did you check how the `file` property shows up in the DSL docs?

The `Signature` class is not part of the DSL docs.

> We also need to add a section to the release notes about possibly breaking changes.

Done in 62d13aecca547272b44f5797e6b5df49c53231b2.